File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) -1
        maxArea = 0
        while left < right:
            maxArea = max(maxArea, (right - left) * min(heights[left], heights[right]))
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
        return maxArea
        

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        char temp1;
        char temp2;
        int left = 0;
        int right = s.size()-1;
        while(left < right){
            // non alpabetric ascii chars
            while(left < right && !isalnum(s[left])){
                left++;
            }
            while(left < right && !isalnum(s[right])){
                right--;
            }

            temp1 = tolower(static_cast<unsigned char>(s[left]));
            temp2 = tolower(static_cast<unsigned char>(s[right]));
            if (temp1 != temp2){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};
